terraform/variables.tf

```diff
@@ -15,6 +15,10 @@
 variable "project_id" {
   description = "The GCP project ID."
   type        = string
+  validation {
+    condition     = length(var.project_id) <= 30
+    error_message = "ERROR: project_id must be <= 30 characters."
+  }
```

Comment:
I'm a little concerned with this:

1. I find client-side validation to be artificially limiting. If GCP ever changed their project IDs to be longer, template users would still be stuck with the old values, and we increase our long-term maintenance. It's legacy, but project IDs can also be prefixed with their org as a syntax that most of the API endpoints accept (e.g. `mydomain.com:my-project`). I'd prefer if we pushed this validation up into the abc CLI, where users have the ability to bypass a validation easily.

2. There are other validations to consider too (must start with a letter, can only contain lowercase letters, digits, and hyphens, cannot end with a hyphen). I'm curious why we're _only_ validating length? Like I said above, I'd strongly prefer if this validation was pushed into the abc cli where we can both (a) centralize the validation and (b) have a good way for power users to bypass the validation.

Reply:
Hmmmmmm, pushing to abc CLI makes most sense, assuming validation is implemented already

Reply:
I think input validation for abc CLI is [in-progress](https://github.com/abcxyz/abc/issues/39#issue-1766105040). Would you prefer I keep or remove the validation in terraform now, until input validation with the CLI is available?

Reply:
I had a sync with dave today, he said its done, maybe sync with him? Rather do it once.